templates/network.py

```diff
@@ -60,23 +60,41 @@ def generate_config(context):
       subnetwork['projectId'] = context.properties['projectId']
 
     resources.append({
-        'name': subnetwork['resourceName'],
-        'type': 'subnetwork.py',
-        'properties': subnetwork
+      'name': subnetwork['resourceName'],
+      'type': 'subnetwork.py',
+      'properties': subnetwork
     })
 
-  return {
-      'resources':
-          resources,
-      'outputs': [{
-          'name': 'name',
-          'value': resource_name
-      }, {
-          'name': 'selfLink',
-          'value': network_self_link
+  if context.properties.get('createCustomStaticRoute', False):
+    resources.append({
+      'name': 'private-google-access-route',
+      # https://cloud.google.com/compute/docs/reference/rest/v1/routes
+      'action': 'gcp-types/compute-v1:compute.routes.insert',
```

Comment:
Should this have a metadata / runtimePolicy / CREATE stanza like the dns_zone action resources? I'm not 100% sure where or when that's needed, but it can't hurt to aim for consistency where possible.

Reply:
So the most I can find for `actions` documentation is from here:
https://github.com/GoogleCloudPlatform/deploymentmanager-samples/tree/master/examples/v2/cloudbuild
which says:
> This sample uses the undocumented DM Actions feature (estimated to come out in Q2 2018) to call the Cloud Container Builder projects.builds.create API. This doesn't create a DM resource like you would normally see, it just calls the API as specified in the metadata.runtimePolicy section. This section may contain the values UPDATE_ALWAYS (call the API for create or update changes in the deployment), CREATE (only call on create), UPDATE_ON_CHANGE (call when the action changes), DELETE (call on deletes).

is `CREATE` sufficient? Or would we want it for update as well?

Reply:
I think CREATE is sufficient – my mental model is that we CREATE a deployment to set up the project, but never do any updates or deletes afterwards.